ReactSkia/components/RSkComponentText.cpp

```diff
@@ -51,84 +52,135 @@ inline int getNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLi
             break;
         }
     }
-    if ((minNumberOfLines > 0) && (minNumberOfLines < maxNumberOfLines))
+    if ((minNumberOfLines) && (maxNumberOfLines) && (minNumberOfLines < maxNumberOfLines))
         return minNumberOfLines;
-    
+        
     return maxNumberOfLines;
 }
 
+inline int getLines(std::vector<LineMetrics>& metrics, Float frameHeight) {
+    int i=0;
+    int minNumberOfLines = 0;
+    for(; i< metrics.size(); i++) {
+        if ((metrics[i]).fBaseline > frameHeight) {
+            minNumberOfLines = (metrics[i]).fLineNumber;
+            break;
+        }
+    }
+    return minNumberOfLines;
+}
+ 
+inline SkPaint convertTextColor(SharedColor textColor) {
```

Comment:
With PT:43 [Props Update], Separate API created for to do color conversion. Reuse this logic to avoid duplication of logic.

Reply:
Updated.